conjur/controller/init_controller.py

```diff
@@ -94,14 +104,15 @@ def _format_conjur_url(self) -> Tuple[str, str]:
 
         return urlparse(self.conjurrc_data.conjur_url)
 
-    def _validate_conjur_url(self, conjur_url: ParseResult, allow_https_only: bool = True):
+    def _validate_conjur_url(self, conjur_url: ParseResult):
         """
         Validates the specified url
 
         Raises a RuntimeError in case of an invalid url format
         """
+        allow_https_only = not self.ssl_verification_data.is_insecure_mode
         valid_scheme = conjur_url.scheme == 'https' or (
-                    not allow_https_only and conjur_url.scheme == 'http')
+                not allow_https_only and conjur_url.scheme == 'http')
         if not valid_scheme:
```

Comment:
this is very unreadable...
Maybe do this instead?
```suggestion
        # http scheme is valid only in insecure mode
        valid_scheme = conjur_url.scheme == 'https'  or self.ssl_verification_data.is_insecure_mode
        if not valid_scheme:
```